### PR TITLE
feat: process & store post questions on post worker processing

### PR DESCRIPTION
**Pull request by DragosIorgulescu**
- always create questions for new posts (if present)
- only update questions for existing posts if there aren't any already present

ref [WT-1866](https://dailydotdev.atlassian.net/browse/WT-1866)

[WT-1866]: https://dailydotdev.atlassian.net/browse/WT-1866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: __tests__/workers/postUpdated.ts
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:118]: No prio, but think this would also work:
```
await repo.save(DEFAULT_QUESTIONS.map((question) => ({ postId, question })));
```

File: __tests__/workers/postUpdated.ts
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:617]: The main before each already has all sources do you need this specific one?

File: __tests__/workers/postUpdated.ts
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:663]: I think maybe do a array check, safer in case the order ever changes

File: __tests__/workers/postUpdated.ts
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:650]: I don't think you need the actual post here, should be able to match if you pass a already mocked post basically.
It would avoid time on database read-writes

File: __tests__/workers/postUpdated.ts
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:667]: This one would already be covered in general as the other test don't include questions.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice work!
Only have some ideas for optimising test cases, otherwise really well done on this, super clean

File: __tests__/workers/postUpdated.ts
Comment (DragosIorgulescu) [reply __tests__/workers/postUpdated.ts]: great call, thank you 🙏

File: __tests__/workers/postUpdated.ts
Comment (DragosIorgulescu) [reply __tests__/workers/postUpdated.ts]: ah, missed that, thanks for pointing it out 👍

File: __tests__/workers/postUpdated.ts
Comment (DragosIorgulescu) [reply __tests__/workers/postUpdated.ts:647]: agreed, but since the other tests don't explicitly test for this, I'd like to still keep this one as a sanity check -- in case the other ones get updated in the future to include questions for some reason

**Review by rebelchris — APPROVED** [review]
Good from my side.

With API projects we always need to wait for @idoshamun approval as well, so don't merge it yet please.

Another note, please see our existing [guidelines](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/375652412/Git#Branch-names) on PR title naming.

**Review by idoshamun — APPROVED** [review]
Perfect!

**Review by capJavert — APPROVED** [review]
Smooth 🚀

**Pull request merged**
